### PR TITLE
onboarding new Lb changes and updating buildspec to fix build issues

### PR DESCRIPTION
**Pull request by godduv-amzn**
*Description of changes:*
Onboarding new Lb changes and updating buildspec to fix build issues

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by naveenrp-amzn — COMMENTED** [review]
Looks good.

**Review by naveenrp-amzn — COMMENTED** [review]
looks good

File: aws-lightsail-loadbalancertlscertificate/src/main/java/software/amazon/lightsail/loadbalancertlscertificate/helpers/GetModifiedLbTlsCertResponse.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package software.amazon.lightsail.loadbalancertlscertificate.helpers;
+
+import software.amazon.awssdk.services.lightsail.model.GetLoadBalancerTlsCertificatesResponse;
+
+/**
+ * We are using the LoadBalancerTlsCertificate resource type to modify the HttpsRedirectionEnabled parameter of the LoadBalancer.
```
Comment (tanbrian) [inline aws-lightsail-loadbalancertlscertificate/src/main/java/software/amazon/lightsail/loadbalancertlscertificate/helpers/GetModifiedLbTlsCertResponse.java:6]: Could you provide more detail as to why we're doing this? It seems weird to use another resource to modify a property that's defined on another resource.

File: buildspec.yml [hunk 2]
```diff
@@ -5,8 +5,11 @@ phases:
       java: openjdk8
       python: 3.7
     commands:
+      -  pip install --upgrade 'pyyaml < 6.0'
       -  pip install --upgrade 'six==1.15.0'
-      -  pip install --upgrade 'pyyaml>=5.4'
+      -  pip install boto3 --upgrade
+      -  pip install --upgrade 'docker~=4.0'
```
Comment (tanbrian) [inline buildspec.yml:11]: Why is the `docker` dependency needed?

**Review by tanbrian — APPROVED** [review]
I've approved the PR in the interest of time, but a couple suggestions for next time:

* Please try to break up the PR into multiple smaller changes if possible. It looks like these PR contains a few unrelated changes (the `tlsPolicy` addition, `httpsRedirectionEnabled` support, and the build fix); creating separate, smaller PRs generally allow them to be reviewed more quickly and thoroughly.
* I don't have much context around these changes, so please include some more info in the PR summary on what the PR is doing, why you've designed it this way, and any other background info for people without context to be able to review the PR.

I'd also strongly recommend reading this info about best practices for writing code reviews: https://google.github.io/eng-practices/review/developer/

File: buildspec.yml (hunk 2 above)
Comment (godduv-amzn) [reply buildspec.yml:11]: This dependency already exists. We just had to change the version to fix the build issue.

**Review by naveenrp-amzn — APPROVED** [review]

**Pull request merged**